src/__tests__/resetpassword.js

```diff
@@ -0,0 +1,293 @@
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+import models from '../models';
+import app from '../index';
+import { sendResetMail, sendSignupMail } from '../services/sendMail';
+import UserController from '../controllers/userController';
+
+// const app = `localhost:${process.env.PORT}`;
+const { expect } = chai;
+chai.use(chaiHttp);
+chai.use(sinonChai);
+
+const { resetPassword } = UserController;
+
+const apiVersion = '/api/v1';
+const forgotPasswordURL = `${apiVersion}/forgotpassword`;
+const resetPasswordURL = `${apiVersion}/resetpassword`;
+const validId = 2;
+const resetToken = '12ererfbuib23iub328o7rg8hbiuva';
+
+// Create table and seed database
+const seedTestDb = async () => {
+  await models.User.create({
+    email: 'youremail@andela.com',
+    role: 'passenger'
+  });
+
+  await models.Login.create({
+    email: 'youremail@andela.com',
+    password: 'password'
+  });
+
+  await models.Reset.create({
+    email: 'youremail@andela.com',
+    password: 'password'
+  });
+};
+
+before(async () => {
+  try {
+    await seedTestDb();
+  } catch (err) {
+    //
+  }
+});
+
+describe('Forgot Password validations', () => {
+  describe('POST /api/forgetpassword', () => {
+    it('should not generate reset link without an email of an existing user', (done) => {
+      chai
+        .request(app)
+        .post(`${forgotPasswordURL}`)
+        .send({})
+        .end((err, res) => {
+          expect(res).to.have.status(400);
+          expect(res.body.status).to.be.equal('error');
```

Comment:
what if the email is valid but the email doesn't exist in the db?

Reply:
User will receive a signup email instead of a reset password message in their email. The code is written as tested as well,